pkg/private/tar/build_tar.py

```diff
@@ -36,27 +37,47 @@ def normpath(path):
   return os.path.normpath(path).replace(os.path.sep, '/')
 
 
+def parse_xattr(xattr_list):
+  xattrs = {}
+  if xattr_list:
+    for item in xattr_list:
+      idx = item.index("=")
+      if idx < 0:
+        raise ValueError("Unexpected xattr item format {} (want key=value)".format(item))
+      key = item[:idx]
+      raw = item[idx+1:]
+      if raw.startswith("0x"):
+        xattrs[key] = bytes.fromhex(raw[2:]).decode('latin-1')
```

Comment:
You probably want to use `.decode('utf-8', "surrogateescape")` here as certain `xattrs` will be re-encoded incorrectly otherwise. On re-encoding the `tarfile` module uses `surrogateescape` to restore any values that don't cleanly decode to UTF-8.